ktor-network/jvm/src/io/ktor/network/sockets/NIOSocket.kt

```diff
@@ -28,9 +28,6 @@ internal abstract class NIOSocketImpl<out S>(
 
     override val socketContext: CompletableJob = Job()
```

Comment:
Not sure if the job should be outside the jobs hierarchy here. Shouldn't we tie it to selector's job?